frame/transaction-payment/Cargo.toml

```diff
@@ -9,6 +9,7 @@ repository = "https://github.com/paritytech/substrate/"
 description = "FRAME pallet to manage transaction payments"
 
 [dependencies]
+serde = { version = "1.0.101", optional = true }
```

Comment:
Why this change?

Reply:
You are right, this is no longer needed. But it _was_ needed when I previously added:

```
pub RuntimeVersion build(|_| true): bool;
```

When I added build, and the module did not have serde, I got the following error:

```
error[E0463]: can't find crate for `serde`
  --> frame/transaction-payment/src/lib.rs:79:1
   |
79 | / decl_storage! {
80 | |     trait Store for Module<T: Trait> as TransactionPayment {
81 | |         pub NextFeeMultiplier get(fn next_fee_multiplier): Multiplier = Multiplier::from_parts(0);
82 | |
83 | |         pub RuntimeVersion build(|_| true): bool;
84 | |     }
85 | | }
   | |_^ can't find crate

error: aborting due to previous error
```

Is this a bug in the macro? Should we auto import serde here if needed?

Reply:
I couldn't find the rustc issue (not sure if there even exists one). However, the problem is that when you have something like this:
```
#[derive(frame_support::serde::Serialize)]
struct Something {}
```

The procedural macro can not see this `frame_support::serde`. Which means it will try to search for `serde` in the current `Cargo.toml`... :( 

So, TLDR, nothing we can do about.